henson-processor/src/test/java/com/f2prateek/dart/henson/processor/IntentBuilderGeneratorTest.java

```diff
@@ -76,7 +76,112 @@
         .generatesSources(builderSource);
   }
 
-  @Test public void hensonNavigable() {
+  @Test
+  public void injectingExtra_withInheritance() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.Test", Joiner.on('\n').join( //
+           "package test;", //
+           "import android.app.Activity;", //
+           "import com.f2prateek.dart.InjectExtra;", //
+           "public class Test extends TestOne {", //
+           "    @InjectExtra String extra;", //
+           "}", //
+           "class TestOne extends Activity {", //
+           "    @InjectExtra String extra2;", //
+           "}" //
+        ));
+
+        JavaFileObject builderSource =
+            JavaFileObjects.forSourceString("test/Test$$IntentBuilder", Joiner.on('\n').join( //
+                "package test;", //
+             "import android.content.Context;", //
+             "import android.content.Intent;", //
+             "import com.f2prateek.dart.henson.Bundler;", //
+             "import java.lang.String;", //
+             "public class Test$$IntentBuilder {", //
+             "  private Intent intent;", //
+             "  private Bundler bundler = Bundler.create();", //
+             "  public Test$$IntentBuilder(Context context) {", //
+             "    intent = new Intent(context, Test.class);", //
+             "  }", //
+             "  public Test$$IntentBuilder.AfterSettingExtra extra(String extra) {", //
+             "    bundler.put(\"extra\", extra);", //
+             "    return new Test$$IntentBuilder.AfterSettingExtra();", //
+             "  }", //
+             "  public class AfterSettingExtra {", //
+             "    public Test$$IntentBuilder.AllSet extra2(String extra2) {", //
+             "      bundler.put(\"extra2\", extra2);", //
+             "      return new Test$$IntentBuilder.AllSet();", //
+             "    }", //
+             "  }", //
+             "  public class AllSet {", //
+             "    public Intent build() {", //
+             "      intent.putExtras(bundler.get());", //
+             "      return intent;", //
+             "    }", //
+             "  }", //
+             "}" //
+        ));
+
+        assert_().about(javaSource())
+                .that(source)
+                .processedWith(ProcessorTestUtilities.hensonProcessorsWithoutParceler())
+                .compilesWithoutError()
+                .and()
+                .generatesSources(builderSource);
+    }
+
+    @Test
+    public void injectingExtra_withPreCompiledInheritance() {
+        JavaFileObject source = JavaFileObjects.forSourceString("test.Test", Joiner.on('\n').join( //
+            "package test;", //
+            "import android.app.Activity;", //
+            "import com.f2prateek.dart.InjectExtra;", //
+            "public class Test extends PreCompiledChildModel {", //
+            "    @InjectExtra String extra;", //
```

Comment:
We should replace this with Extra3 so that we should also have the super super parent's extra. Now it's masked.